### ISSUE TITLE
Feature Request: Autocomplete/DropDown should be navigatable by keyboard (accessibility)

### ISSUE THREAD

**Issue by rugk** [grocy/grocy#2404]
## STR

1. E.g. go to /product/new to create a new product.
2. Enter something as a parent product beginning with something you already have.
3. Now try pressing the down key on your keyboard to select the entry.

## What happens

This is how it looks:
![grafik](https://github.com/grocy/grocy/assets/11966684/8b93bca7-e5a5-4bf2-ae40-e66b6d7cdc56)

Down key does not select anything.

### What should happen

This is an accessibility and usability issue. Things to improve:
1. Forms should in general be navigable by keyboard only.
2. The contrast in the dark mode is very bad, you can hardly even notice this is a dopdown there. Some border/shadow etc. is missing.
3. I guess the element does not use standard HTML elements and nowadays [many are there](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/optgroup). This is an accessibility issue for screen readers etc.

Best idea is to use a GUI framework or something that does all this out-of-the-box.

**Comment by berrnd** [comment]
> Enter something as a parent product beginning with something you already have.
> Now try pressing the down key on your keyboard to select the entry.

Did that now, did that yesterday (and most likely kind of everyday since Grocy exists), works:

https://github.com/grocy/grocy/assets/2161815/ca6ac49f-cb31-450c-8a02-73867b8ea3b4


&nbsp;
> Forms should in general be navigable by keyboard only.

They generally are.

&nbsp;
> The contrast in the dark mode is very bad

Night mode was contributed, I only barely maintain it and design choices in general are at the end always and mostly something that depend on your personal taste. So feel free to tweak [`grocy_night_mode.css`](https://github.com/grocy/grocy/blob/master/public/css/grocy_night_mode.css) for everyone or simply use [custom CSS](https://github.com/grocy/grocy#adding-your-own-css-or-js-without-to-have-to-modify-the-application-itself) to make everything look your very own way of how stuff should be designed.

&nbsp;
> I guess the element does not use standard HTML elements and nowadays [many are there](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/optgroup).

Thanks for that suggestion, never heard of HTML before - since it sounds you're an expert: Which of those standard HTML elements provides all the functionality (an auto-complete dropdown with additional/hidden search data per item) needed? I agree that the currently used component, choosen more than 6 years ago, is not longer the best choice and I already wanted to replace and unify all dropdowns throughout the frontend by a better one, but currently my desire to work on this project is fairly limited.

&nbsp;
> This is an accessibility issue for screen readers etc.

Grocy's frontend is not optimized for screen readers at all (and it's also nowhere stated that that's the case) and I have no plans to change that, mostly due to it's just a lot of work - when I work on this project, I prefer to do the fun stuff, since this is a hobby.

&nbsp;
> Best idea is to use a GUI framework or something that does all this out-of-the-box.

Sounds again you're a top-notch expert, so what about that suggestion: Grocy has [an API](https://demo-prerelease.grocy.info/api), why no start building the next generation frontend you've ever (at least since yesterday, that day you started using Grocy and already found the biggest security vulnerability 🤣🤣🤣 of all time) dreamed of right tonight?

**Comment by rugk** [comment]
Again no need to be passive aggressive and attack other. You can take advise or leave it. Just please refrain from personal attacks, thanks.

So let's focus on the key down part. I see from your video it does work on Windows.

I currently don't feel welcomed in this community to contribute, however. Otherwise I would have looked into why it does not work on Linux as wanted.
